answerking_app/utils/model_types.py

```diff
@@ -1,42 +1,63 @@
-from typing import Any
+import datetime
 
 from typing_extensions import (  # for Python <3.11 with (Not)Required
     NotRequired,
     TypedDict,
 )
+from typing import Any
 
 
-class ItemType(TypedDict):
+class ProductType(TypedDict):
     id: NotRequired[int]
     name: str
-    price: str
+    price: str | float
     description: str
+    categories: NotRequired["list[CategoryType]"]
     retired: NotRequired[bool]
-    stock: int
-    calories: int
+
+
+class CategoryProductType(TypedDict):
+    id: int
 
 
 class CategoryType(TypedDict):
     id: NotRequired[int]
     name: NotRequired[str]
+    description: NotRequired[str]
+    createdOn: NotRequired[datetime.datetime | str]
+    lastUpdated: NotRequired[datetime.datetime | str]
     retired: NotRequired[bool]
-    items: NotRequired["list[ItemType]"]
+    products: NotRequired["list[CategoryProductType]"]
 
 
-class OrderItemType(TypedDict):
+class OrderProductType(TypedDict):
     id: NotRequired[int]
-    name: NotRequired[str]
-    price: NotRequired[str]
     quantity: int
     sub_total: NotRequired[str]
 
 
 class OrderType(TypedDict):
     id: NotRequired[int]
-    address: NotRequired[str]
-    order_items: NotRequired["list[OrderItemType]"]
-    status: NotRequired[str]
-    total: NotRequired[str]
+    order_status: NotRequired[str]
+    order_total: NotRequired[str]
+    created_on: NotRequired[datetime.datetime]
+    last_updated: NotRequired[datetime.datetime]
+    line_items: NotRequired["list[OrderProductType]"]
+
+
+class OrderProductTypeApiFormat(TypedDict):
+    product: ProductType
+    quantity: int
+    subTotal: NotRequired[float]
+
+
+class OrderTypeApiFormat(TypedDict):
+    id: NotRequired[int]
+    orderStatus: NotRequired[str]
+    orderTotal: NotRequired[float]
```

Comment:
Decimal not float

Reply:
it's what we get from the API, we changed that from string, to float if I'm not wrong